classes/AlertView/Standard.php

```diff
@@ -0,0 +1,391 @@
+            $this->data['alertsearch_pretty'] = $alertsearch_pretty;
+        }
+
+        if ( $this->data['alertsearch'] ) {
+            $this->data['search_text'] = $this->data['alertsearch'];
+        } else {
+            $this->data['search_text'] = $this->data['keyword'];
+        }
+    }
+
+    private function prettifyCriteria($alert_criteria) {
+        $text = '';
+        if ( $alert_criteria ) {
+            $criteria = explode(' ', $alert_criteria);
+            $words = array();
+            $spokenby = '';
+
+            foreach ($criteria as $c) {
+                if (preg_match('#^speaker:(\d+)#',$c,$m)) {
+                    $MEMBER = new \MEMBER(array('person_id'=>$m[1]));
+                    $spokenby = $MEMBER->full_name();
+                } else {
+                    $words[] = $c;
+                }
+            }
+            if ( $spokenby && count($words) ) {
+                $text = $spokenby . ' mentions [' . implode(' ', $words) . ']';
+            } else if ( count( $words ) ) {
+                $text = '[' . implode(' ', $words) . ']' . ' is mentioned';
+            } else if ( $spokenby ) {
+                $text = "$spokenby speaks";
+            }
+
+            return $text;
+        }
+
+        return $text;
+    }
+
+    private function checkForCommonMistakes() {
+        $mistakes = array();
+        if (strstr($this->data['alertsearch'], ',') > -1) {
+            $mistakes['multiple'] = 1;
+        }
+
+        if (
+                preg_match('#([A-Z]{1,2}\d+[A-Z]? ?\d[A-Z]{2})#i', $this->data['alertsearch'], $m) &&
+                strlen($this->data['alertsearch']) > strlen($m[1]) &&
+                validate_postcode($m[1])
+        ) {
+            $this->data['postcode'] = $m[1];
+            $this->data['scottish_text'] = '';
+            $this->data['mp_display_text'] = '';
+            if (postcode_is_scottish($m[1])) {
+                $this->data['mp_display_text'] = 'your MP, ';
+                $this->data['scottish_text'] = ' or MSP';
+            }
+            $mistakes['postcode_and'] = 1;
+        }
+
+        $this->data['mistakes'] = $mistakes;
+    }
+
+    private function formatSearchMemberData() {
+        if ( isset($this->data['postcode']) ) {
+            try {
+                $postcode = $this->data['postcode'];
+
+                $MEMBER = new \MEMBER( array('postcode' => $postcode) );
+                // move the postcode to the front just to be tidy
+                $tidy_alertsearch = $postcode . " " . trim(str_replace("$postcode", "", $this->data['alertsearch']));
+                $alertsearch_display = str_replace("$postcode ", "", $tidy_alertsearch);
+
+                $this->data['member_alertsearch'] = str_replace("$postcode", "speaker:" . $MEMBER->person_id, $tidy_alertsearch);
+                $this->data['member_displaysearch'] = $alertsearch_display;
+                $this->data['member'] = $MEMBER;
+
+                if ( $this->data['scottish_text'] ) {
```

Comment:
Hmm, the postcode shortcutting means this is no longer reached if you enter a Scottish postcode, it jumps to assuming you want MP. Currently this doesn't matter given we don't have Scottish updating, but just to note as and when that's fixed we might want to not shortcut Scottish postcodes so that this bit of code runs and lets you pick.